File: examples/pizzas_domain/lib/pizzas_domain/pizza/events/added_topping.rb
module PizzasDomain
  class Pizza
    module Events
      class AddedTopping
        attr_reader :aggregate_id, :pizza, :name, :amount, :description, :toppings, :occurred_at

        def initialize(aggregate_id: nil, pizza: nil, name: nil, amount: nil, description: nil, toppings: nil)
          @aggregate_id = aggregate_id
          @pizza = pizza
          @name = name
          @amount = amount
          @description = description
          @toppings = toppings
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: pizzas_static_go/domain/added_topping.go
package domain

import "time"

type AddedTopping struct {
	AggregateID string `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	Name string `json:"name"`
	Amount int64 `json:"amount"`
	Description string `json:"description"`
	Toppings []Topping `json:"toppings"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (a AddedTopping) EventName() string { return "AddedTopping" }

func (a AddedTopping) GetOccurredAt() time.Time { return e.OccurredAt }
